tests/utils/logging_tester.py

```diff
@@ -0,0 +1,79 @@
+from functools import wraps
+import os
+import json
+
+
+def _check_logs(reference_logfile, generated_logfile):
+    reference_logs = open(reference_logfile, "r").readlines()
+    generated_logs = open(generated_logfile, "r").readlines()
+
+    if len(reference_logs) != len(generated_logs):
+        msg_assert = ""
+        if len(reference_logs) > len(generated_logs):
+            diff = len(reference_logs) - len(generated_logs)
+            msg_assert += f"Logs generated are SHORTER than the expected logs. There are {diff} extra logs.\n"
+            msg_assert += "Last log of the generated log is : \n"
+            msg_assert += generated_logs[-1]
+        else:
+            diff = len(generated_logs) - len(reference_logs)
+            msg_assert += f"Logs generated are LONGER than the expected logs.\n There are {diff} extra logs :\n"
+            for log in generated_logs[len(reference_logs) :]:
+                msg_assert += log
+        assert 0, msg_assert
+
+    for index, (ref, gen) in enumerate(zip(reference_logs, generated_logs)):
+        # As they are string, we only need to check if they are equal. If they are not, we then compute a more precise difference, to debug.
+        if ref == gen:
+            continue
+        ref_log = json.loads(ref)
+        gen_log = json.loads(gen)
+        diff_keys = [
+            d1[0] for d1, d2 in zip(ref_log.items(), gen_log.items()) if d1[1] != d2[1]
+        ]
+        # \n does not work in f-strings.
+        newline = "\n"
+        assert (
+            len(diff_keys) == 0
+        ), f"Lgos don't match at {index} st log. : \n{newline.join([f'In {key} field, got -> {gen_log[key]}. {newline}Expected : -> {ref_log[key]}.' for key in diff_keys])}"
```

Comment:
Comment: I know it's black and not you, but I _hate_ this indentation.

Also typos "**Lgos** don't match at {} **st** log."

Reply:
Also it'd be nice if these two lines lined up one on top of the other:

`In {key} field god -> `
`Expected ->`

You can try to do line them up by introducing `\t` chars in the second one